src/metabase/api/user.clj

```diff
@@ -88,12 +89,13 @@
   ;; can't change email if it's already taken BY ANOTHER ACCOUNT
   (api/checkp (not (db/exists? User, :email email, :id [:not= id]))
     "email" "Email address already associated to another user.")
-  (api/check-500 (db/update-non-nil-keys! User id
-                   :email        email
-                   :first_name   first_name
-                   :last_name    last_name
-                   :is_superuser (when (:is_superuser @api/*current-user*)
-                                   is_superuser)))
+  (api/check-500 (db/update! User id
+                   (-> body
+                       (u/select-keys-when :present #{:login_attributes}
+                                           :non-nil (set (concat [:first_name :last_name]
+                                                                 (when (:is_superuser @api/*current-user*)
+                                                                   [:is_superuser]))))
+                       (assoc :email email))))
```

Comment:
I thing while we're at it we might as well just make `email` an option API param and then stick `:email` in the `:non-nil` list with the other ones

That way you can update just the name of a User for example without having to pass in the whole object. More of `PATCH`-style operation than traditional `PUT`